include/Dialect/PolyExt/IR/PolyExtOps.td

```diff
@@ -0,0 +1,43 @@
+#ifndef HEIR_INCLUDE_DIALECT_POLYEXT_IR_POLYEXTOPS_TD_
+#define HEIR_INCLUDE_DIALECT_POLYEXT_IR_POLYEXTOPS_TD_
+
+include "PolyExtDialect.td"
+include "include/Dialect/Poly/IR/PolyDialect.td"
+include "include/Dialect/Poly/IR/PolyTypes.td"
+include "mlir/IR/OpBase.td"
+include "mlir/Interfaces/InferTypeOpInterface.td"
+include "mlir/Interfaces/SideEffectInterfaces.td"
+
+
+class PolyExt_Op<string mnemonic, list<Trait> traits = []> :
+        Op<PolyExt_Dialect, mnemonic, traits> {
+  let cppNamespace = "::mlir::heir::poly_ext";
+}
+
+def PolyExt_CModSwitchOp : PolyExt_Op<"cmod_switch", traits = [Pure, ElementwiseMappable]> {
+  let summary = "Move polynomial into a ring with different `cmod` by scaling polynomial coefficients proportionally";
+
+  let description = [{
+    `poly_ext.cmod_switch` "moves" polinomial $x$ from ring $Z/QZ[X]/(f(x))$ to the ring $Z/qZ[X]/(f(x))$.
+    The move is done by scaling the polynomial coefficients by $q/Q$ and "rounding" so input and output coufficients are congruent modulo `congruence_modulus` $t$.
```

Comment:
Remind me what the purpose of the second modulus $t$ is? cc @asraa

Reply:
`t` is the actual plaintext modulus, while `Q` and `q` are the ring moduli - for these schemes you can project onto a smaller ring and thus scale the noise by the ratio of `Q` and `q`. For correctness you require that this operation keeps the values equivalent modulo the plaintext modulus. Often, the moduli are chosen to be the same mod `t`, but otherwise you need to add a small delta (the rounding) to correct for that.